modules/biobank/php/collection.class.inc

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types=1);
+/**
+ * Specimen Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\biobank;
+
+/**
+ * Specimen Object Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Object instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class Collection extends Process
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a specimen entity.
+     *
+     * @access private
+     */
+    private $quantity; //float
+    private $unitId;   //int
+
+    /**
+     * Sets the current quantity of specimen available. This associated unit
+     * can be found in the `biobank_unit` using the unit ID.
+     *
+     * @param float $quantity the current amount of specimen available
+     *
+     * @return void
+     */
+    public function setQuantity(float $quantity) : void
+    {
+        $this->quantity = $quantity;
+    }
+
+    /**
+     * Gets the current amount of specimen available
+     *
+     * @return float
+     */
+    public function getQuantity() : float
+    {
+        return $this->quantity;
+    }
+
+    /**
+     * Sets the ID of the specimen's unit of quantity
```

Comment:
I'm not sure what this means.. is it an ID which is a foreign key to something? If so, why not create/pass a Unit class instead of an integer?

Reply:
It's an ID which references the unit. It's indeed a foreign key to the unit table which facilitates mapping to the unit label. 

The reason we are not passing a unit class is because it was decided that Unit does not justify having it's own class because it doesn't qualify as an entity. Secondly, even if Unit were an entity, we decided that we are simply passing IDs to be able to map to the entity, rather than the entire entity to avoid infinite chains of entity loading. 

We can discuss this in more detail at our Loris entity meetings.

@ridz1208 @xlecours @johnsaigle